src/databricks/labs/ucx/assessment/export.py

```diff
@@ -0,0 +1,116 @@
+import os
+import re
+import csv
+import logging
+from pathlib import Path
+from zipfile import ZipFile
+from concurrent.futures import ThreadPoolExecutor
+from databricks.labs.blueprint.tui import Prompts
+from databricks.labs.ucx.contexts.workspace_cli import WorkspaceContext
+
+logger = logging.getLogger(__name__)
+
+
+class Exporter:
+    # File and Path Constants
+    _ZIP_FILE_NAME = "ucx_assessment_results.zip"
+
+    def __init__(self, ctx: WorkspaceContext):
+        self._ctx = ctx
+
+    def _get_ucx_main_queries(self) -> list[dict[str, str]]:
+        """Retrieve and construct the main UCX queries."""
+        pattern = r"\b.inventory\b"
+        schema = self._ctx.inventory_database
+        project_root = Path(__file__).parent.parent.parent.parent
+        ucx_main_queries_path = project_root / "labs/ucx/queries/assessment/main"
```

Comment:
this is a way concise version of this class that does the same thing, but with portability to become part of https://github.com/databrickslabs/lsql. You can PR it over there first and call it from UCX after, if you'd like. This way we can export any dashboards-as-code into CSV for any project.

```python
from databricks.labs.lsql.dashboards import DashboardMetadata

dashboard = DashboardMetadata.from_path(ucx_main_queries_path)
dashboard = dashboard.replace_database(catalog='hive_metastore', database=self._config.inventory_database)
for tile in dashboard.tiles:
   if not tile.is_query():
      continue
   file_name = f"{tile.id}.csv"
   for row in self._sql_backend.fetch(tile.content):
      _ = row.as_dict()
```

Reply:
WIP